File: src/1152-Analyze-User-Website-Visit-Pattern/1152.js
var mostVisitedPattern = function(username, timestamp, website) {
    var data = new Map();
    var n = username.length;
    for (var i = 0; i < n; ++i) {
        for (var j = 0; j < n; ++j) {
            if (timestamp[i] < timestamp[j] 
               && username[i] == username[j])
            for (var k = 0; k < n; ++k) {
                if (timestamp[j] < timestamp[k]
                   && username[j] == username[k])
                {
                    var t = website[i] + "," + website[j] + "," + website[k];
                    if (!data.has(t)) data.set(t, new Set())
                    data.get(t).add(username[k]);
                }
            }
        }
    }
    var ta = Array.from(data);
    ta.sort(function(a, b) {
        if (a[1].size == b[1].size) return a[0].localeCompare(b[0]);
        return  b[1].size - a[1].size;
    });
    return ta[0][0].split(",");
};

File: src/1152-Analyze-User-Website-Visit-Pattern/1152.go
func mostVisitedPattern(username []string, timestamp []int, website []string) []string {
    data, n := make(map[string]map[string]int), len(username)
    for i := 0; i < n; i++ {
        for j := 0; j < n; j++ {
            if timestamp[i] < timestamp[j] && username[i] == username[j] {
                for k := 0; k < n; k++ {
                    if timestamp[j] < timestamp[k] && username[j] == username[k] {
                        ts := website[i] + "," + website[j] + "," + website[k]
                        if _, ok := data[ts]; !ok {
                            data[ts] = make(map[string]int)
                        }
                        data[ts][username[k]]++
                    }     
                }
            }
        }
    }
    var s1 []string
    for k := range data {
        s1 = append(s1, k)
    }
    sort.Strings(s1)
    res, s := "", 0
    for _, v := range s1 {
        if len(data[v]) > s {
            s = len(data[v])
            res = v
        }
    }
    return strings.Split(res, ",")
}